app/assets/javascripts/angular/filters/filters.js

```diff
@@ -12,17 +12,23 @@ angular.module("Prometheus.filters").filter('toPercentile', function() {
 
 angular.module("Prometheus.filters").filter('hostnameFqdn', function() {
   return function(input) {
-    var a = document.createElement("a");
-    a.href = input;
-    return a.host;
+    var re = /^([^:\/?]+:\/\/(www\.)?)?([^\/]+)/;
+    var match = input.match(re);
+    if (match && match[3]) {
+      return match[3].replace('www.', '').replace(/:[0-9]+/,'');
+    }
+    return '';
   };
 });
 
 angular.module("Prometheus.filters").filter('hostname', function() {
   return function(input) {
-    var a = document.createElement("a");
-    a.href = input;
-    return a.host.split(".", 1)[0];
+    var re = /^([^:\/?]+:\/\/(www\.)?)?([^\/]+)/;
+    var match = input.match(re);
+    if (match && match[3]) {
+      return match[3].replace('www.', '').replace(/:[0-9]+/,'').split('.')[0];
+    }
```

Comment:
Could we just call the hostname function here, instead of duplicating the code?

pseudo code:

```
$filter('hostnameFqdn')(input).split('.')[0]
```